tests/run/fused_types.pyx

```diff
@@ -401,6 +401,40 @@ def test_cython_numeric(cython.numeric arg):
     """
     print cython.typeof(arg), arg
 
+
+cdef fused int_t:
+    int
+
+cdef _test_pylong(int_t i):
+    print cython.typeof(i)
+
+def test_pylong(int_t i):
+    """
+    >>> import cython
+    >>> try:    long = long # Python 2
+    ... except: long = int  # Python 3
+
+    >>> test_pylong[int](int(0))
+    int
+    >>> test_pylong[cython.int](int(0))
+    int
+    >>> test_pylong(int(0))
+    int
+
+    >>> test_pylong[int](long(0))
+    int
+    >>> test_pylong[cython.int](long(0))
+    int
+    >>> test_pylong(long(0))
+    int
+
+    >>> test_pylong[cython.long](0)  # doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    KeyError: ...
+    """
+    _test_pylong(i)
```

Comment:
I don't think the `cdef` function is of any use here. There is no internal dispatch since both functions use the same fused type.
```suggestion
def test_pylong(int_t i):
    """
    >>> import cython
    >>> try:    long = long # Python 2
    ... except: long = int  # Python 3

    >>> test_pylong[int](int(0))
    int
    >>> test_pylong[cython.int](int(0))
    int
    >>> test_pylong(int(0))
    int

    >>> test_pylong[int](long(0))
    int
    >>> test_pylong[cython.int](long(0))
    int
    >>> test_pylong(long(0))
    int

    >>> test_pylong[cython.long](0)  # doctest: +ELLIPSIS
    Traceback (most recent call last):
    KeyError: ...
    """
    print(cython.typeof(i))
```